components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -2439,4 +2442,59 @@ private boolean isAccessTokenExpired(long issuedTimeInMillis, long validityPerio
 
         return OAuth2Util.getTimeToExpire(issuedTimeInMillis, validityPeriodMillis) < 0;
     }
+
+    public Set<AccessTokenDO> getAccessTokensByBindingRef(AuthenticatedUser user, String bindingRef)
+            throws IdentityOAuth2Exception {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Retrieving active access tokens issued to user, " + user.getUserName() + " with binding " +
+                    "reference " + bindingRef);
+        }
+
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+        try {
+            String sqlQuery = OAuth2Util.getTokenPartitionedSqlByUserId(SQLQueries
+                    .GET_ACCESS_TOKENS_BY_BINDING_REFERENCE, user.getUserName());
+            int tenantId = OAuth2Util.getTenantId(user.getTenantDomain());
+            Map<String, AccessTokenDO> tokenMap = new HashMap<>();
+            jdbcTemplate.executeQuery(sqlQuery,
+                    (resultSet, i) -> {
+                        String token = resultSet.getString("ACCESS_TOKEN");
+                        AccessTokenDO accessTokenDO = new AccessTokenDO();
+                        if (tokenMap.containsKey(token)) {
+                            AccessTokenDO tokenObj = tokenMap.get(token);
+                            String[] previousScope = tokenObj.getScope();
+                            String[] newSope = new String[tokenObj.getScope().length + 1];
+                            System.arraycopy(previousScope, 0, newSope, 0, previousScope.length);
+                            newSope[previousScope.length] = resultSet.getString(2);
+                            tokenObj.setScope(newSope);
+                        } else {
+                            String consumerKey = resultSet.getString("CONSUMER_KEY");
+                            String tokenScope = resultSet.getString("TOKEN_SCOPE");
+                            String refreshToken = resultSet.getString("REFRESH_TOKEN");
+                            String tokenId = resultSet.getString("TOKEN_ID");
+
+                            String[] scope = OAuth2Util.buildScopeArray(tokenScope);
+                            accessTokenDO.setAccessToken(token);
+                            accessTokenDO.setConsumerKey(consumerKey);
+                            accessTokenDO.setScope(scope);
+                            accessTokenDO.setAuthzUser(user);
+                            accessTokenDO.setTenantID(tenantId);
+                            accessTokenDO.setRefreshToken(refreshToken);
+                            accessTokenDO.setTokenId(tokenId);
```

Comment:
Are we using all of these values which are set to the accessTokenDO

Reply:
We need the token, consumer key and scope.